api/src/main/java/org/ehcache/event/CacheEvent.java

```diff
@@ -69,4 +69,72 @@
    */
   @Deprecated
   Cache<K, V> getSource();
+
+  /**
+   * This marks events fireable atomically
+   */
+  void markFireable();
```

Comment:
That's a no go - all these methods can't make it to a pubic API. I believe the `ThreadLocal` approach is going to make some not needed, but they really need to all go - use a wrapper where you need these.